test/IRGen/keypaths_c_types.swift

```diff
@@ -0,0 +1,8 @@
+// RUN: %target-swift-frontend -primary-file %s -emit-ir -import-objc-header %S/Inputs/keypaths_c_types.h
+
+// This used to crash while trying to emit a reference to the property
+// descriptor for the some_field property.
```

Comment:
Should we be emitting property descriptors for imported types?

Reply:
No, but we did try to reference it because we had a key path that included an external decl for the imported type and the IRGen would try to emit a reference to the property descriptor here:

https://github.com/apple/swift/blob/8437ee5df089081a4f76b565db188b30ba33e498/lib/IRGen/GenKeyPath.cpp#L899

Reply:
Right, I'm suggesting that we could in theory also emit property descriptors for computed properties of imported types. We'll need to do this so that equality and hashing of keypaths that reference imported types works across translation units -- we need to unique the descriptor at runtime, just like we do for foreign type metadata.

This looks like a fine fix in the interim, though. Do you mind filing a bug for implementing the runtime uniqueing of imported property descriptors? CC @jckarter who might have a better idea.

Reply:
The property descriptor is for resilience, not uniquing. For imported properties, we'd probably need to use a string for the uniquing ID.